src/components/Settings/Settings.tsx

```diff
@@ -49,7 +51,23 @@ const Settings = () => {
 
   // Submits the form. Triggered by pressing the save button
   const handleSubmit: FormEventHandler = (e) => {
-    e.preventDefault();
+    // e.preventDefault();
+    if (activeNode) {
+      const updatedUser: User = {
+        id: deviceID,
+        longName: deviceName,
+        shortName: deviceNickname,
+        macaddr: activeNode ? activeNode.data.user!.macaddr : [],
+        hwModel: activeNode ? activeNode.data.user!.hwModel : 0,
+        isLicensed: activeNode ? activeNode.data.user!.isLicensed : true,
```

Comment:
Should this default to true without an active node?

Reply:
This legally cannot default to true, blocking PR on this. I wouldn't recommend completely rebuilding the `User` object, I would get the current device if it exists (if not I wouldn't even open the popup), then pull the user from that device's corresponding node. Lmk if you wanna chat on this

Reply:
Removed the ternary - the default would never have been used, and I only put it there because I thought it would throw a type error. Will  look into pulling the current `User`.